crates/diesel_models/src/query/role.rs

```diff
@@ -170,4 +199,96 @@ impl Role {
             },
         }
     }
+
+    pub async fn generic_list_roles_by_entity_type(
+        conn: &PgPooledConn,
+        entity_type: ListRolesByEntityPayload,
+        is_lineage_data_required: bool,
+        limit: Option<u32>,
+    ) -> StorageResult<Vec<Self>> {
+        let mut query = <Self as HasTable>::table().into_boxed();
+
+        match entity_type {
+            ListRolesByEntityPayload::Organization(org_id) => {
+                let entity_in_vec = if is_lineage_data_required {
+                    vec![
+                        EntityType::Organization,
+                        EntityType::Merchant,
+                        EntityType::Profile,
+                    ]
+                } else {
+                    vec![EntityType::Organization]
+                };
+                query = query
+                    .filter(dsl::org_id.eq(org_id))
+                    .filter(
+                        dsl::scope
+                            .eq(RoleScope::Organization)
+                            .or(dsl::scope.eq(RoleScope::Merchant))
+                            .or(dsl::scope.eq(RoleScope::Profile)),
```

Comment:
Should we make scopes also a Vec? Will that improve readability?

Converting to Vec will give us an advantage, we don't have to change the query if scopes are added or removed. But I assume, we won't be changing scopes anytime soon.

Reply:
updated the query with the discussed changes